### PR TITLE
Add commitment routes

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: sdk/src/client/node_api/core/routes.rs [hunk 1]
```diff
@@ -349,6 +348,56 @@ impl ClientInner {
...
+    pub async fn get_slot_commitment_by_id_index(&self, slot_index: &SlotIndex) -> Result<Vec<u8>> {
+        let path = &format!("api/core/v3/commitments/by-index/{slot_index}");
+
+        self.node_manager
+            .read()
+            .await
+            .get_request_bytes(path, None, self.get_timeout().await)
+            .await
+    }
+
```
Comment (Thoralf-M) [inline sdk/src/client/node_api/core/routes.rs:400]: What about `/api/core/v3/commitments/by-index/{index}/utxo-changes` and`/api/core/v3/commitments/{commitmentId}/utxo-changes`?

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs:400]: Will do another PR

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/client/node_api/core/routes.rs [hunk 2]
```diff
@@ -349,6 +348,56 @@ impl ClientInner {
...
+        self.node_manager
+            .read()
+            .await
+            .get_request::<SlotCommitment>(path, None, self.get_timeout().await, false, true)
+            .await
+    }
+
+    /// Gets the slot commitment, as raw bytes, by the given slot index.
+    /// /api/core/v3/commitments/by-index/{index}
+    pub async fn get_slot_commitment_by_id_index(&self, slot_index: &SlotIndex) -> Result<Vec<u8>> {
```
Comment (qrayven) [inline sdk/src/client/node_api/core/routes.rs]: shouldn't be `by_index_raw`?

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Good catch

File: sdk/src/client/node_api/core/routes.rs [hunk 3]
```diff
@@ -349,6 +348,56 @@ impl ClientInner {
...
     }
 
+    // Commitments routes.
+
+    /// Gets the slot commitment by the given slot commitment id.
+    /// GET /api/core/v3/commitments/{commitmentId}
+    pub async fn get_slot_commitment_by_id(&self, slot_commitment_id: &SlotCommitmentId) -> Result<SlotCommitment> {
+        let path = &format!("api/core/v3/commitments/{slot_commitment_id}");
+
+        self.node_manager
```
Comment (qrayven) [inline sdk/src/client/node_api/core/routes.rs:358]: can we just reuse the `get_slot_commitment_by_id_raw`?

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs:358]: We could but I'd rather not, not sure we have the guarantee the raw version will exist on all nodes, etc...

Comment (qrayven) [reply sdk/src/client/node_api/core/routes.rs:358]: aa ok makes sense

File: sdk/src/client/node_api/core/routes.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Fixed

File: sdk/src/client/node_api/core/routes.rs [hunk 4]
```diff
@@ -349,6 +348,56 @@ impl ClientInner {
...
+            .read()
+            .await
+            .get_request::<SlotCommitment>(path, None, self.get_timeout().await, false, true)
+            .await
+    }
+
+    /// Gets the slot commitment, as raw bytes, by the given slot index.
+    /// /api/core/v3/commitments/by-index/{index}
```
Comment (Alex6323) [inline sdk/src/client/node_api/core/routes.rs]: totally nit-picky...so ignore if you want ... but here you missed the `GET`, also you use `index` in the doc, but `slot_index` would be better I think as you do in the actual route.

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: I'm adding the GET but I'd rather keep the rest identical to the API spec

File: sdk/src/types/block/slot/index.rs [hunk 5]
```diff
@@ -10,6 +10,12 @@ use derive_more::{Deref, From};
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct SlotIndex(u64);
 
+impl core::fmt::Display for SlotIndex {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
```
Comment (Alex6323) [inline sdk/src/types/block/slot/index.rs]: you can just `derive_more` this I think

File: sdk/src/client/node_api/core/routes.rs (hunk 4 above)
Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Done

File: sdk/src/types/block/slot/index.rs (hunk 5 above)
Comment (thibault-martinez) [reply sdk/src/types/block/slot/index.rs]: Done

**Review by qrayven — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**